src/Components/Bookmark/Bookmark.js

```diff
@@ -1,92 +1,41 @@
-import React from 'react';
+import {useEffect,useState} from 'react';
 import { MenuBar, SuggestionBar } from "../index";
+import { useDispatch, useSelector } from "react-redux";
+import { SinglePost } from "../Home/SinglePost";
+
 
 function Bookmark() {
+
+const [bookmarkedPosts , setBookmarkedPosts] = useState([])
+const { allPost } = useSelector((state) => state.post);
```

Comment:
you can also do like this `const allPost = useSelector(state => state.post.allPost` as you need only one key from the state object.